switches/http2/http/HttpDispatcher.cpp

```diff
@@ -273,15 +289,25 @@ namespace Aseba
 	
 	void HttpDispatcher::getApiDocs(HandlerContext& context)
 	{
-		json response;
+		json response = R"({"swagger":"2.0","schemes":["http"],"host":"localhost:3000","info":{"version":"1","title":"Aseba","description":"REST API for Aseba"},"parameters":{"trait:serverSentEventStream:todo":{"name":"todo","in":"query","required":false,"type":"integer"}},"responses":{"trait:serverSentEventStream:200":{"description":"stream of server-sent events","schema":{"type":"string"}}}})"_json;
 		for (const auto& handlerMapKV: handlers)
 		{
 			for (const auto& handlerKV: handlerMapKV.second)
 			{
-				response.push_back({
-					{ "method", toString(handlerMapKV.first) },
-					{ "uri", handlerKV.first }
-				});
+				if (handlerKV.first.size() == 0)
+					break;
+				// build uri string with slash delimiters
+				std::stringstream uri;
+				for (auto elt: handlerKV.first) // copy so can modify
+					uri << "/" << elt;
+				// method key must be lower case for OAS 2.0
+				auto method = toString(handlerMapKV.first);
+				std::transform(method.begin(), method.end(), method.begin(), ::tolower);
+
+				// apidoc for this operation should have been registered
+				json doc = apidocs[handlerMapKV.first][handlerKV.first];
+				// patch this operation into the apidocs
+				response["paths"][uri.str()][method] = doc;
```

Comment:
The complete api-docs object needs to be constructed from a basic JSON object, with general information about the API (keys `host`, `info`, `schemes`, `swagger`), model definitions (key `definitions`), and then an entry under key `path` for every endpoint.

So I first create the JSON object (line 292, in this version without the `definitions`), then insert the endpoint in line 310. The comment says "patch" because I was thinking of JSON Patch ([RFC 6902](https://tools.ietf.org/html/rfc6902)), but in fact it is just an insertion.